crates/analyzer-host/src/lsp_impl/state.rs

```diff
@@ -127,4 +171,104 @@ impl State {
 	pub(crate) fn set_progress(&mut self, progress_manager: ProgressManager) {
 		self.progress_manager = Some(progress_manager);
 	}
+
+	async fn process_background_analyze_requests(
```

Comment:
I would split this function up by moving `analyze_source_text` out of it, and making the `async move {...}` block on line 207 it's own function

Reply:
When you say 'block', what do you mean?

Reply:
Block of code is any code in-between an open and closed brackets, [example for C++](https://web1.eng.famu.fsu.edu/~haik/met.dir/hcpp.dir/notes.dir/cppnotes/node38.html#:~:text=In%20C%2B%2B%2C%20a%20code,opening%20and%20closing%20curly%20braces.&text=The%20four%20statements%20after%20the,represent%20a%20block%20of%20code.). In this case it would be line 198 to 226 but it looks like you've already reduced the block size.

Reply:
Ah, sorry. I misunderstood. I thought you were talking about async or concurrency because it just that way in my head.